app/backend/src/couchers/servicers/references.py

```diff
@@ -231,48 +232,52 @@ def AvailableWriteReferences(self, request, context):
+            ).scalar_one_or_none():
                 context.abort(grpc.StatusCode.NOT_FOUND, errors.USER_NOT_FOUND)
 
             can_write_friend_reference = (
-                session.query(Reference)
-                .filter(Reference.from_user_id == context.user_id)
-                .filter(Reference.to_user_id == request.to_user_id)
-                .filter(Reference.reference_type == ReferenceType.friend)
-                .one_or_none()
+                session.execute(
+                    select(Reference)
+                    .where(Reference.from_user_id == context.user_id)
+                    .where(Reference.to_user_id == request.to_user_id)
+                    .where(Reference.reference_type == ReferenceType.friend)
+                ).scalar_one_or_none()
             ) is None
 
             q1 = (
-                session.query(literal(True), HostRequest)
+                select(literal(True), HostRequest)
                 .outerjoin(
                     Reference,
                     and_(
                         Reference.host_request_id == HostRequest.conversation_id,
                         Reference.from_user_id == context.user_id,
                     ),
                 )
-                .filter(Reference.id == None)
-                .filter(HostRequest.can_write_reference)
-                .filter(HostRequest.from_user_id == context.user_id)
-                .filter(HostRequest.to_user_id == request.to_user_id)
+                .where(Reference.id == None)
+                .where(HostRequest.can_write_reference)
+                .where(HostRequest.from_user_id == context.user_id)
+                .where(HostRequest.to_user_id == request.to_user_id)
             )
 
             q2 = (
-                session.query(literal(False), HostRequest)
+                select(literal(False), HostRequest)
                 .outerjoin(
                     Reference,
                     and_(
                         Reference.host_request_id == HostRequest.conversation_id,
                         Reference.from_user_id == context.user_id,
                     ),
                 )
-                .filter(Reference.id == None)
-                .filter(HostRequest.can_write_reference)
-                .filter(HostRequest.from_user_id == request.to_user_id)
-                .filter(HostRequest.to_user_id == context.user_id)
+                .where(Reference.id == None)
+                .where(HostRequest.can_write_reference)
+                .where(HostRequest.from_user_id == request.to_user_id)
+                .where(HostRequest.to_user_id == context.user_id)
             )
 
-            host_request_references = q1.union_all(q2).order_by(HostRequest.end_time_to_write_reference.asc()).all()
+            union = union_all(q1, q2).order_by(HostRequest.end_time_to_write_reference.asc()).subquery()
+            host_request_references = session.execute(select(union.c[0], aliased(HostRequest, union))).all()
```

Comment:
Is `.c[0]` the first column? Are you able to do something like naming it (I think it's `label('whatever')`?), then `.c.whatever`?

Reply:
Yes, it's the first column

`union_all` returns a tuple of values, no objects, hence the need for aliased. I can perhaps add a line to add a name? It doesn't add much, but it makes it a little clearer:

```
            union = union_all(q1, q2).order_by(HostRequest.end_time_to_write_reference.asc()).subquery()
            union = select(union.c[0].label("surfed"), aliased(HostRequest, union))
            host_request_references = session.execute(union).all()
```

Reply:
Yeah I think that's a bit clearer.

Also, what's the `aliased` doing here? I'm not sure I understand

Reply:
It reconstitutes the object

https://github.com/sqlalchemy/sqlalchemy/discussions/6714

Reply:
Thanks!